app/models/driver.rb

```diff
@@ -0,0 +1,38 @@
+class Driver < ApplicationRecord
+  has_many :trips, dependent: :nullify
+
+  validates :name, presence: true
+  validates :vin, uniqueness: true
+
+  def rating
+    num_trips = self.trips.length
+    ongoing_trips = 0
+    rating = 0
+
+    num_trips.times do |i|
+      if self.trips[i].rating != nil
+        rating += self.trips[i].rating
+      else
+        ongoing_trips += 1
+      end
+    end
+
+    rating = rating.to_f/(num_trips-ongoing_trips)
+    return rating.round(2)
+  end
+
+  def earned
+    num_trips = self.trips.length
+    money = 0
+
+    num_trips.times do |i|
```

Comment:
Instead of using a `times` loop, it may be better to think of this problem as iterating through the collection of `self.trips`, so I would prefer maybe to use an `each` loop instead.